druid/src/contexts.rs

```diff
@@ -443,12 +443,25 @@ impl_context_method!(EventCtx<'_, '_>, UpdateCtx<'_, '_>, LifeCycleCtx<'_, '_>,
     // requirement. Needs type ids recorded. This goes wrong if you don't have a pod between you and a lens.
     pub fn new_sub_window<W: Widget<U> + 'static, U: Data>(
         &mut self,
-        window_config: WindowConfig,
+        mut window_config: WindowConfig,
         widget: W,
         data: U,
         env: Env,
     ) -> WindowId {
         trace!("new_sub_window");
+
+        if matches!(
+            window_config.level,
+            Some(WindowLevel::DropDown(_) | WindowLevel::Tooltip(_))
```

Comment:
According to the `WindowLevel` docs the difference between various levels is only in their z-order, so I don't think you should change behavior based on `WindowLevel`.

Reply:
> According to the `WindowLevel` docs the difference between various levels is only in their z-order, so I don't think you should change behavior based on `WindowLevel`.

Well, the docs also say: 

> Describes the purpose of a window and should be mapped appropriately to match platform conventions.

WindowLevel is also used to set the WM type hint, so this is by sure more than a Z-order.

Anyways, I cant see any other property which indicates the window type, so what exactly do you have in mind?

Reply:
That's a good point, although it isn't particularly consistent with the previous sentence. I guess we need to figure out (and document!) what else `WindowLevel` is actually supposed to mean.

But I still think it would be surprising if window positioning worked differently for different kinds of windows: why should it be that tooltips are relative to the parent's content but modals are relative to the parent's window decorations?

Reply:
> But I still think it would be surprising if window positioning worked differently for different kinds of windows: why should it be that tooltips are relative to the parent's content but modals are relative to the parent's window decorations?

Agreed. That is why is suggested to use two different `get_position` methods (`get_window_position` and `get_position`).

Reply:
> Agreed. That is why is suggested to use two different `get_position` methods (`get_window_position` and `get_position`).

Please ignore that last argument - it does not really makes any difference.

> > But I still think it would be surprising if window positioning worked differently for different kinds of windows: why should it be that tooltips are relative to the parent's content but modals are relative to the parent's window decorations?

Thought a bit more about that, and it is quite obvious that dropdowns and tooltips need to be positioned relative to the content root.

Reply:
I agree that it's very natural for dropdowns and tooltips to be positioned relative to the content. But there are also some disadvantages:
- it's undocumented
- it differs from established APIs, like `gtk_window_move` and (I think) `SetWindowPos` on windows, which are always relative to the window frame
- if we support other type hints in the future, we will need to decide (and document!) how they're treated

The first point is the big one: if we're going to go down this road, it needs to be clearly documented.